server/src/main/java/org/eclipse/lsp/cobol/core/model/variables/NodeConverter.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.model.variables;
+
+import com.google.common.collect.ImmutableList;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.model.tree.Node;
+import org.eclipse.lsp.cobol.core.model.tree.NodeType;
+import org.eclipse.lsp.cobol.core.model.tree.variables.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The call converts Variable node to Variable for backward compatibility.
+ * This is temporal converter and must be removed at the end of SyntaxTree migration.
+ */
+public class NodeConverter {
+  private Map<Long, Variable> convertedVariables = new HashMap<>();
+
+  /**
+   * Convert Variable node into the variable.
+   *
+   * @param variableNode the variable tree node
+   * @return a variable
+   */
+  public Variable convertVariable(VariableNode variableNode) {
+    Variable variable = convert(variableNode);
+    convertedVariables.put(variableNode.getId(), variable);
+    return variable;
+  }
+
+  private Variable convert(VariableNode variableNode) {
+    switch (variableNode.getVariableType()) {
+      case CONDITION_DATA_NAME: return convert((ConditionDataNameNode) variableNode);
+      case ELEMENT_ITEM: return convert((ElementItemNode) variableNode);
+      case GROUP_ITEM: return convert((GroupItemNode) variableNode);
+      case INDEPENDENT_DATA_ITEM: return convert((IndependentDataItemNode) variableNode);
+      case INDEX_ITEM: return convert((IndexItemNode) variableNode);
+      case MNEMONIC_NAME: return convert((MnemonicNameNode) variableNode);
+      case MULTI_TABLE_DATA_NAME: return convert((MultiTableDataNameNode) variableNode);
+      case RENAME_ITEM: return convert((RenameItemNode) variableNode);
+      case TABLE_DATA_NAME: return convert((TableDataNameNode) variableNode);
+      default: throw new RuntimeException("Unexpected variable type");
```

Comment:
I am afraid that Coveritry may complain about the general exception type. You may use `IllegalArgumentException` here